tutorials/klusterai-api/model-comparison.ipynb

```diff
@@ -1,659 +1,909 @@
+        "ax.bar_label(bars, label_type='center', color='white', fmt=\"%.3f\")\n",
+        "\n",
+        "# Add token usage as text above bars\n",
+        "for bar, model in zip(bars, accuracies.keys()):\n",
+        "    prompt_tokens = token_usage[model]['total_prompt_tokens']\n",
+        "    completion_tokens = token_usage[model]['total_completion_tokens']\n",
+        "    cost = token_usage[model]['total_cost_usd']\n",
+        "\n",
+        "    # Positioning text above the bar\n",
+        "    height = bar.get_height()\n",
+        "    ax.text(\n",
+        "        bar.get_x() + bar.get_width() / 2,\n",
+        "        height + 0.005,\n",
+        "        f\"Input tokens: {prompt_tokens}\\nOutput tokens: {completion_tokens}\\nTotal tokens: {prompt_tokens+completion_tokens}\\nCost: ${cost}\",\n",
+        "        ha='center',\n",
+        "        va='bottom',\n",
+        "        fontsize=10,\n",
+        "        color='black'\n",
+        "    )\n",
+        "\n",
+        "# Labels and title\n",
+        "ax.set_ylim(0, 1)\n",
+        "ax.set_xlabel('Model')\n",
+        "ax.set_ylabel('Accuracy')\n",
+        "ax.set_title('Accuracy by model with token usage')\n",
+        "\n",
+        "plt.tight_layout()\n",
+        "plt.show()"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "id": "eed7c5fe-ad80-4a73-81e4-e2ece7c59f1d",
+      "metadata": {},
+      "source": [
+        "The analysis shows that Gemma 3 27B achieved the highest classification accuracy at 82.9%, making it the top-performing model on this task. Despite consuming the largest number of tokens, its flat-rate pricing model helped it maintain a low overall cost ($0.051), placing it as the second cheapest model in the evaluation. This balance of performance and affordability makes it a strong candidate for accuracy-sensitive applications where moderate cost is acceptable.\n",
+        "\n",
+        "Llama 4 Maverick 17B stood out as the most cost-effective model. It delivered a solid 80.5% accuracy while incurring the lowest total cost of just $0.0316 — nearly half the cost of Gemma 3. For use cases that prioritize efficiency, scalability, and cost-sensitivity (e.g., high-volume inference or real-time pipelines), Maverick offers exceptional value.\n",
+        "\n",
+        "Llama 4 Scout 17B showed worse accuracy (79.8%) than Maverick but at a higher price point ($0.1165). This model might be appropriate in other scenarios where it can yield better results, such as tasks involving ambiguity or subtle classification differences. However, its cost-efficiency ratio lags behind both Maverick and Gemma.\n",
+        "\n",
+        "DeepSeek V3 0324 similar accuracy of both Maverick and Scout, while also incurring the highest cost ($0.0992), suggesting that in this specific context, it may not be the most competitive choice when weighed against either cost or performance alone. \n",
```

Comment:
```suggestion
        "DeepSeek V3 0324 had a similar accuracy to both Maverick and Scout, while also incurring the highest cost ($0.0992), suggesting that in this specific context, it may not be the most competitive choice when weighed against either cost or performance alone. \n",
```